cv_next/app/profile/[profileUsername]/components/profileDisplay.tsx

```diff
@@ -0,0 +1,35 @@
+import { ProfileKeys } from "@/lib/supabase-definitions";
+import Categories from "@/types/models/categories";
+
+export const ProfileDisplay = ({ user }: { user: UserModel }) => {
+  return (
+    <>
+      <div className="m-auto w-full">
+        <div className="flex flex-wrap justify-between">
+          <label className="font-bold" htmlFor="username">
+            Username:{" "}
+          </label>
+          <span id="username">{user.full_name}</span>
```

Comment:
Now you should have two field displays - both the username and the display name. See e.g. github:
![image](https://github.com/user-attachments/assets/14159083-1d2b-4528-9693-bfe006236135)

Reply:
That is not what we talked about, at this point we use username as a client replacement to uuid so it remains
constant as generated for the time being. If we would like to implement such a change, it would be on the next version. if and when we do a db overhall

Reply:
Not sure I understand.
Username is a unique id that can be used for client-side usage instead of the uuid.
The display name/full_name is the displayable name of the user (to be used in comments, cvs etc).
What is there to implement further (aside from what you've already done)?

Reply:
so let me understand you want to also display on profile the generated username?

Reply:
instead of displaying and editing the displayname only

Reply:
OK, two points:
1. Without regarding point 2, the full name is display name, and not username. Let's keep this terminology throughout the project. So if you're displaying the full name (=display name), the label should be "display name".
2. Remember that we require the user to change the generated username - it's only generated for first usage. Perhaps I've missed a discussion, but from what you're saying it seems that you've decided to only let the user change their display name, and not their username? If so, why? In most sites where you have both display name and username, you can change both attributes..

Reply:
in #202 we agreed to make only the display name editable in order to use the generated username to use as replacement for user id in client side. And so it will not change instead there is a new flow in `app/callback`
we use `validateUsername` to regenerate if its null and when we generate it we also change the display name temporarily to fit. and the user is able to change the display name

Reply:
maybe there was a miscommunication but I showed it in the weekly meeting and @Aradon267 was pleased

Reply:
OK I'm sorry I might have not explained myself properly.
What I meant is the following:
We agreed on having a display name property (that I said we don't need to be a new one, but simply a rename of full_name). When I said to make it editable I didn't mean *instead* of the username, but rather *in addition to*, because username, I believe, should stay editable. It's still a visible property of the user, so the user should be able to edit it as they please (as long as it stays unique).
@Aradon267 additional thoughts?

Reply:
ok so I understand the confusion, and that fine by me. but just so were on the same page it will cause `URI problems` because we it to fetch the user in profile

Reply:
also it means adding a new field in the unified editing of profile data

Reply:
Adding a new field - I agree. But it shouldn't be more than 1-2 lines of codes iirc.
Causing URI problems - I agree as well, that's why I left a comment on the username update that it'll require a bit more complicated logic than simple update.
Honestly, I'm not 100% against removing that option - but I believe that in every site where you have a username, you generally also have an option to edit it. So it makes sense that users will ask for it.
And frankly, we use the user's id in all the server logic, so changing the username won't cause any issue other than the profile page. If you want, I can sketch a solution.

Reply:
I tell you what if we were to show the display name as the username wouldn't that will solve your problem?

It's like the old discord model you have your 'discord tag' and the username that you are able to change (In our case the 'display_name' is the username, and the tag is the 'generated username'). We could even go as far as regenerating the tag when the 'display_name' changes.

Sounds good?

Reply:
Just to make sure that I understood @lordYorden the discord thing will leave the tag(the generated field) uneditable by the user if it's not via regenerating based on display name changes? I don't know If that's preferable than the username and display name being changeable(based on experience with for example how LinkedIn handles their user links)

Reply:
I agree with Arad, also the issue with relying on display name for basically anything other than profile page (not saying that it isn't useful there - it's very useful there, especially in this kind of project) is distinguishability. Not saying that there's a high chance for it to happen - but the chance 100% exists, that in a single comments section we'll have two people with the same name.